src/cross_section.F90

```diff
@@ -55,32 +54,23 @@ subroutine calculate_xs(p)
       ! Determine if this material has S(a,b) tables
       check_sab = (mat % n_sab > 0)
 
-      ! Initialize position in i_sab_nuclides
-      j = 1
-
       ! Add contribution from each nuclide in material
       do i = 1, mat % n_nuclides
         ! ========================================================================
         ! CHECK FOR S(A,B) TABLE
 
         i_sab = 0
 
-        ! Check if this nuclide matches one of the S(a,b) tables specified -- this
-        ! relies on i_sab_nuclides being in sorted order
+        ! Check if this nuclide matches one of the S(a,b) tables specified
         if (check_sab) then
-          if (i == mat % i_sab_nuclides(j)) then
+          if (mat % nuc_sab_dict % has_key(i)) then
```

Comment:
Although I like the simplicity of the data structure (and the fact that it removes 40 net lines of code), under the hood this line here could be quite a bit more expensive than the simple comparison it is replacing. And given that it is called in a performance-sensitive routine (calculate_xs), I'm not sure whether it is the best idea. I haven't done any comparisons to back it up, but it does worry me.